components/brave_shields/browser/brave_shields_util.cc

```diff
@@ -438,28 +438,39 @@ void DispatchBlockedEvent(const GURL& request_url,
 #endif
 }
 
-bool ShouldSetReferrer(bool allow_referrers,
-                       bool shields_up,
-                       const GURL& original_referrer,
-                       const GURL& tab_origin,
-                       const GURL& target_url,
-                       const GURL& new_referrer_url,
-                       network::mojom::ReferrerPolicy policy,
-                       Referrer* output_referrer) {
-  if (!output_referrer || allow_referrers || !shields_up ||
-      original_referrer.is_empty() ||
-      // Same TLD+1 whouldn't set the referrer
-      SameDomainOrHost(
-          target_url, original_referrer,
-          net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES) ||
-      // Whitelisted referrers shoud never set the referrer
-      (g_brave_browser_process &&
-       g_brave_browser_process->referrer_whitelist_service()->IsWhitelisted(
-           tab_origin, target_url.GetOrigin()))) {
+bool MaybeChangeReferrer(bool allow_referrers,
+                         bool shields_up,
+                         const GURL& current_referrer,
+                         const GURL& tab_origin,
+                         const GURL& target_url,
+                         network::mojom::ReferrerPolicy policy,
+                         Referrer* output_referrer) {
+  DCHECK(output_referrer);
+  if (allow_referrers || !shields_up || current_referrer.is_empty()) {
     return false;
   }
+
+  // TODO(iefremov): Get rid of the whitelist once our webcompat is OK.
```

Comment:
I guess that will also get rid of the layering violation for g_brave_browser_process so I won't worry about it here